src/Http/Http.Extensions/src/JsonOptions.cs

```diff
@@ -23,8 +23,8 @@ public class JsonOptions
 
         // The JsonSerializerOptions.GetTypeInfo method is called directly and needs a defined resolver
         // setting the default resolver (reflection-based) but the user can overwrite it directly or by modifying
-        // the TypeInfoResolverChain
-        TypeInfoResolver = JsonSerializer.IsReflectionEnabledByDefault ? CreateDefaultTypeResolver() : null
+        // the TypeInfoResolverChain. Use JsonTypeInfoResolver.Combine() to produce an empty TypeInfoResolver.
+        TypeInfoResolver = JsonSerializer.IsReflectionEnabledByDefault ? CreateDefaultTypeResolver() : JsonTypeInfoResolver.Combine()
```

Comment:
When trying to serialize with `PublishTrimmed == true` this will throw a "No metadata for type 'Foo'". Consider authoring a custom resolver that throws a desirable error message. You could also have STJ do it for you by setting the value to `JsonSerializerOptions.Default.TypeInfoResolver` (although that would require a suppresion).